cirq-google/cirq_google/engine/engine_result_test.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+from typing import Mapping
+
+import pandas as pd
+
+import cirq
+import cirq_google as cg
+import numpy as np
+
+
+def test_engine_result():
+    res = cg.EngineResult(
+        job_id='my_job_id',
+        job_finished_time=datetime.datetime(2022, 4, 1, 1, 23, 45),
```

Comment:
This naive datetime will not be serializable, right? In that case I'd suggest adding a timezone, so we model best practices. Or maybe we should support serializing naive datetimes :-)

Reply:
In the constructor of `cg.EngineResult`, I canonicalize everything to "aware" utc datetimes, so this will still function. 

Do you think we should only allow "aware" datetimes in EngineResult? Or just take whatever is input and face an error on serialization? 

In any event; I'll change the test fixtures to use aware datetimes.

Reply:
It's probably worth a broader discussion how we want to handle datetimes. My personal opinion would be to allow naive and aware datetimes in serialization as well, since we can convert either to a timestamp in a well-defined way, just like we can normalize them in the `EngineResult` constructor. I agree for ergonomics it's nice to handle either when constructing `EngineResult`, but just as it's potentially surprising to not be able to do round-trip equality if you serialize a naive datetime and get back an aware one, it also feels suprising to change things when contructing an object:
```python
dt = datetime.datetime.now()
res = cg.EngineResult(job_finished_time=dt, ...)
dt == res.job_finished_time  # False!
```
That's not _such_ a big deal here since people will generally not be constructing `EngineResult` instances manually.

Reply:
Yes, I agree with everything you say. Let me see if I can write down the options in a way everyone can agree on and then we can pick one.

Reply:
see https://github.com/quantumlib/Cirq/issues/5301

Reply:
I've removed the constructor-conversion from this PR. I recommend if we relax the serialization constraints we do that in a follow-up PR